tests/tracer/telemetry/test_writer.py

```diff
@@ -124,6 +126,107 @@ def test_app_started_event(mock_time, mock_send_request, telemetry_writer):
+
+    def assert_app_heartbeat_event(seq_id):
+        assert len(httpretty.latest_requests()) == seq_id
+        request = httpretty.last_request()
+        assert request.headers["DD-Telemetry-Request-Type"] == "app-heartbeat"
+        assert request.parsed_body == _get_request_body({}, "app-heartbeat", seq_id=seq_id)
+
+    # Assert clean slate
+    assert len(httpretty.latest_requests()) == 0
+
+    # Assert first flush does not queue any events
+    telemetry_writer.periodic()
+    assert len(httpretty.latest_requests()) == 0
+
+    # Advance time 60 seconds
+    mock_time.return_value += 60
+
+    # Assert next flush contains app-heartbeat event
+    telemetry_writer.periodic()
+    assert_app_heartbeat_event(1)
+
+    # Advance time 59 seconds
+    mock_time.return_value += 59
+
+    # Assert next flush does not contain an app-heartbeat event
+    telemetry_writer.periodic()
+    assert len(httpretty.latest_requests()) == 1
+
+    # Advance time 1 second
+    mock_time.return_value += 1
+
+    # Assert next flush contains app-heartbeat event
+    telemetry_writer.periodic()
+    assert_app_heartbeat_event(2)
+
+    # Advance time 120 seconds (2 intervals)
+    mock_time.return_value += 120
+
+    # Assert next flush contains only one app-heartbeat event
+    telemetry_writer.periodic()
+    assert_app_heartbeat_event(3)
+
+
+def test_app_heartbeat_event(mock_time, mock_send_request, telemetry_writer):
+    # type: (mock.Mock, Any, TelemetryWriter) -> None
+    """asserts that we queue/send app-heartbeat event every 60 seconds when app_heartbeat_event() is called"""
+
+    def assert_app_heartbeat_event(seq_id):
+        assert len(httpretty.latest_requests()) == seq_id
+        request = httpretty.last_request()
+        assert request.headers["DD-Telemetry-Request-Type"] == "app-heartbeat"
+        assert request.parsed_body == _get_request_body({}, "app-heartbeat", seq_id=seq_id)
+
+    # Assert clean slate
+    assert len(httpretty.latest_requests()) == 0
+
+    # Within 60 seconds of telemetry writer creation, no event is queued
+    telemetry_writer.app_heartbeat_event()
+    telemetry_writer.periodic()
+    assert len(httpretty.latest_requests()) == 0
+
+    # Advance time 60 seconds
+    mock_time.return_value += 60
+
+    # Manuall queueing and calling periodic will only queue and flush 1 event
+    telemetry_writer.app_heartbeat_event()
+    telemetry_writer.periodic()
+    assert_app_heartbeat_event(1)
+
+    # Advance time 120 seconds (2 intervals)
+    mock_time.return_value += 120
+
+    # Calling multiple times will only enqueue one event
+    telemetry_writer.app_heartbeat_event()
+    telemetry_writer.app_heartbeat_event()
+    telemetry_writer.app_heartbeat_event()
+    telemetry_writer.app_heartbeat_event()
+    telemetry_writer.periodic()
+    assert_app_heartbeat_event(2)
+
+
+def test_app_heartbeat_event_fork(mock_time, telemetry_writer):
```

Comment:
this test is written the same way as the other fork tests, but the other fork tests are wrong and aren't working properly.

these need `mock_send_request`, and when added we get assertion errors, but because it happens in a fork the pytest test doesn't get marked as a failure.

as well, because of how these tests are setup, the fork hooks aren't actually registered either, so they will still send the requests on fork that we are expecting not to send. they only pass because we aren't actually mocking the http calls.